lib/puma/thread_pool.rb

```diff
@@ -298,27 +301,26 @@ def shutdown(timeout=-1)
         # Wait for threads to finish without force shutdown.
         threads.each(&:join)
       else
-        # Wait for threads to finish after n attempts (+timeout+).
-        # If threads are still running, it will forcefully kill them.
-        timeout.times do
-          threads.delete_if do |t|
-            t.join 1
-          end
-
-          if threads.empty?
-            break
-          else
-            sleep 1
+        join = ->(timeout) do
```

Comment:
I don't understand why this proc is created and then called a few lines later.

Reply:
Is it because you want to call it twice with the same logic?

Reply:
I thought it was kind of interesting way to avoid creating a method (which would not have been called anywhere else).

Reply:
Yeah, it's just a lightweight form of local code reuse to avoid repeating the same few lines of code 3 times in a row.